src/util/protection.cpp

```diff
@@ -0,0 +1,88 @@
+	bool is_crash_object(rage::joaat_t model)
+	{
+		if (!model_info::get_model(model))
+			return false;
+		if (!model_info::is_model_of_type(model, eModelType::Object, eModelType::Time, eModelType::Weapon, eModelType::Destructable, eModelType::WorldObject, eModelType::Sprinkler, eModelType::Unk65, eModelType::Plant, eModelType::LOD, eModelType::Unk132, eModelType::Building))
+			return true;
+		for (auto iterator : crash_objects)
+			if (iterator == model)
+				return true;
+		return false;
+	}
+
+	constexpr uint32_t crash_peds[] = {RAGE_JOAAT("slod_human"), RAGE_JOAAT("slod_small_quadped"), RAGE_JOAAT("slod_large_quadped")};
+	bool is_crash_ped(rage::joaat_t model)
+	{
+		for (auto iterator : crash_peds)
+			if (iterator == model)
+				return true;
+		if (!model_info::is_model_of_type(model, eModelType::Ped, eModelType::OnlineOnlyPed))
+			return true;
+		return false;
+	}
+
+	constexpr auto crash_vehicles = {RAGE_JOAAT("arbitergt"), RAGE_JOAAT("astron2"), RAGE_JOAAT("cyclone2"), RAGE_JOAAT("ignus2"), RAGE_JOAAT("s95")};
+	bool is_crash_vehicle(rage::joaat_t model)
+	{
+		for (auto iterator : crash_vehicles)
+			if (iterator == model)
+				return true;
+		if (!model_info::is_model_of_type(model, eModelType::Vehicle, eModelType::Unk133))
+			return true;
+		return false;
+	}
+
+	constexpr uint32_t valid_player_models[] = {
```

Comment:
Change this to a constexpr auto without brackets just like crash_vehicles for consistency, and to allow the compiler to figure it out on its own, also chose to either have a giant one-line list, or each model on a new line for every list on the sheet here.